### ISSUE TITLE
Keyboard Navigation to a disabled donate button allows you to click the link

### ISSUE THREAD

**Issue by dimitur2204** [podkrepi-bg/frontend#1327]
### Describe the Bug

When you use your keyboard to go to one of the finished campaigns and click the donate link it redirects you to the page that should generally be inaccessible.

### To Reproduce

1. Go to the homepage
2. Start hitting Tab on your keybaord until you get to one of the finished donations
3. Get to the disabled Donate button and hit enter
4. It will redirect you to the page

![keyboard-nav-donate-link-bug](https://user-images.githubusercontent.com/61479393/217199020-5363c57d-5ca8-4210-b39f-ab309721b4fe.gif)

### Expected Behavior

It should be untabbable with a `tabindex={0}` 

### Which browser are you using? (if relevant)

Not relevant.

**Comment by ani-kalpachka** [comment]
Have in mind that according to the new design of the Home page (shown here: https://github.com/podkrepi-bg/frontend/issues/1302) we will not have a Donate button on the successful campaign cards, but this is a well spotted defect and it will be good if we fix it until the new design is implemented since currently it is not in progress.

**Comment by dimitur2204** [comment]
> Have in mind that according to the new design of the Home page (shown here: #1302) we will not have a Donate button on the successful campaign cards, but this is a well spotted defect and it will be good if we fix it until the new design is implemented since currently it is not in progress.

Good to know we can reference this in the new design issue as a "Keep in mind" thing and close it if the Homepage closes before it is fixed

**Comment by sashko9807** [comment]
In the current usecase of LinkButton component the [Button component](https://github.com/podkrepi-bg/frontend/blob/master/src/components/common/LinkButton.tsx#L19) provided by `@mui/material` is essentially an anchor([ref](https://mui.com/material-ui/api/button/#props)).  In that case legacyBehavior prop needs to be passed to the Link component as stated by the [Link  docs](https://nextjs.org/docs/api-reference/next/link#if-the-child-is-a-tag). 

Without the legacyBehavior prop what we end up with the following HTML structure
`<a href="...." style="pointer-events: disabled">`
`<button></button>`
`</a>` 
 In which the button is actually disabled, but the `<a href=""></a>` remains valid, since `pointer-events` is not meant to work for keyboard navigation.

Only issue is that custom style should be provided for focused elements, otherwise you wont be able to see which LinkButton is focused. Here is a working example if you want to test

```javascript
const LinkButton = (
  { href, as, prefetch, locale, disabled, ...props }: NextLinkProps,
  ref: Ref<LinkRef>,
) => (
  <Link
    href={href}
    as={as}
    prefetch={prefetch}
    locale={locale}
    passHref
    legacyBehavior>
    <Button ref={ref} disabled={disabled} {...props} sx={{
        "&.Mui-focusVisible": {
      border: 2,
      borderColor: 'orange'
    },
    }}/>
  </Link>
)
````

Solves [#1328 ](https://github.com/podkrepi-bg/frontend/issues/1328) as well.
